internal/app/management/run.go

```diff
@@ -0,0 +1,456 @@
+package management
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lawrencegripper/ion/internal/app/management/module"
+	"github.com/rs/xid"
+	log "github.com/sirupsen/logrus"
+	context "golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type server struct{}
+
+// Kubernetes client
+type Kubernetes struct {
+	client                    *kubernetes.Clientset
+	namespace                 string
+	AzureSPSecretRef          string
+	AzureBlobStorageSecretRef string
+	AzureServiceBusSecretRef  string
+	MongoDBSecretRef          string
+}
+
+// Management holds metadata used by the management server
+type Management struct {
+	DispatcherImageName string
+	DispatcherImageTag  string
+	ID                  string
+}
+
+var k Kubernetes
+var mgmt *Management
+var sharedServicesSecretName string
+var sharedImagePullSecretName string
+var logLevel string
+
+func genID() string {
+	id := xid.New()
+	return id.String()
+}
+
+// Run the GRPC server
+func Run(config *Configuration) {
+
+	logLevel = config.LogLevel
+
+	mgmt = &Management{
+		ID:                  genID(),
+		DispatcherImageName: config.DispatcherImage,
+		DispatcherImageTag:  config.DispatcherImageTag,
+	}
+
+	var err error
+	k.client, err = getClientSet()
+	if err != nil {
+		panic(fmt.Errorf("error connecting to Kubernetes %+v", err))
+	}
+	k.namespace = config.Namespace
+
+	err = createSharedServicesSecret(config)
+	if err != nil {
+		panic(err)
```

Comment:
Would this cause the management API to panic if it was restarted? I believe "create" would fail as the secret would already exist but could be wrong. Maybe would be best to do an update if on start if the items already existed or would that be confusing if admins made changes to the secrets? Possible best to noop if a secret of that name already exists.